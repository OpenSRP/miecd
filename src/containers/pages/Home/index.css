.welcome-box {
    margin-top: 75px;
}

.home-main .components-list {
    display: flex;
    align-items: center;
    justify-content: center;
    padding-left: 25px;
    position: relative;
    flex-direction: row;
    height: 400px;
}

.center-vertically {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100%;
}

.nav {
    background-color: rgb(246, 254, 255);
    max-width: 25%;
    min-width: 140px;
    flex: 1;
    font-family: 'Roboto', 'Helvetica';
}

.main-container {
    display: flex;
    font-family: 'Roboto', 'Helvetica';
    font-size: 16px;
    background: white;
}

.main-container .content {
    height: 100%;
    position: relative;
    background: #f8fafe;
}

.home-main {
    display: flex;
    flex-direction: column;
    padding: 0 50px 0 50px;
    background: white;
    height: 100vh;
    position: relative;
}

.home-main .welcome-text {
    justify-self: flex-start;
    text-align: center;
    margin-top: 2rem;
    margin-bottom: 2rem;
}

.home-main .welcome-text h1 {
    color: #1b314b;
    font-size: 32px;
    font-weight: 500;
    letter-spacing: -0.33px;
    line-height: 38px;
    text-align: center;
}

.home-main .button-style {
    border-radius: 24px;
    border: 1px solid #4d54b4;
    background: initial;
    border-radius: 24px;
    width: 100%;
    color: #4d54b4;
}

.home-main .button-style.default {
    border-radius: 24px;
    width: 100%;
    color: #ffffff;
    font-family: Roboto;
    letter-spacing: 0.1px;
    color: #4d54b4;
    height: 40px;
}

.home-main .button-style.default:hover {
    color: #ffffff;
    background: #4d54b4;
}

.sub-container-heading {
    border: none;
    font-size: 22px;
    text-align: left;
}

.home-main .sub-container-message {
    display: flex;
    flex-direction: column;
    font-size: 1rem;
    height: 100%;
    width: 90%;
    transform: translateY(-20%);
}

.home-main .spacer {
    height: 100%;
    width: 30%;
    flex: 1 1 auto;
    margin: 25px;
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: #f7f6fb;
    border: 1px solid #dadce0;
    border-radius: 18px;
    box-shadow: 0 1px 3px 0 #d7d7e0;
    padding-top: 10px;
}

.spacer > img {
    display: flex;
    width: 14rem;
    height: 100%;
    object-fit: cover;
    transform: translateY(-50%);
}

h4 p {
    text-align: left;
}

.home-main #cont-size {
    display: flex;
    height: auto;
    color: #637381;
}

.home-main #cont-size h1 {
    color: #001429;
    font-size: 24px;
    font-weight: 500;
    letter-spacing: -0.27px;
    line-height: 29px;
    text-align: left;
}

.home-main #cont-sized button:focus {
    outline: none;
}

.home-main #cont-size h4 {
    color: #637381;
    font-size: 18px;
    font-weight: 500;
    letter-spacing: 0.1px;
    line-height: 22px;
}

.home-main #cont-sized {
    top: 20%;
    background: initial;
    position: relative;
    text-align: center;
}

@media screen and (max-width: 1100px) {
    .home-main {
        height: auto;
    }
    .home-main .components-list {
        flex-direction: column;
        height: auto;
        padding: 0;
    }
    .home-main .spacer {
        width: 100%;
        height: 400px;
        margin: 60px 20px;
    }
}
