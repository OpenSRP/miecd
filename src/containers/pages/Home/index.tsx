// this is the home page component
import * as React from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { Link } from 'react-router-dom';
import pregnancyAvatar from '../../../assets/images/pregnancy.jpeg';
import nbcPncAvatar from '../../../assets/images/pnc_nbc.jpeg';
import nutritionAvatar from '../../../assets/images/nutrition.jpeg';
import { NBC_AND_PNC_URL, NUTRITION_URL, PREGNANCY_URL } from '../../../constants';
import './index.css';

class Home extends React.Component<WithTranslation> {
    public render() {
        const { t } = this.props;
        return (
            <div className="home-main">
                <div className="welcome-text">
                    <h1>{t('Welcome to the MIECD dashboard')}</h1>
                </div>
                <div className="center-vertically">
                    <div className="components-list">
                        <div className="spacer">
                            <img src={pregnancyAvatar} id="pregnancyAvatar" alt="pregnancy module avatar" />
                            <div className="sub-container-message">
                                <div id="cont-size">
                                    <h1>{t('Pregnancy')}</h1>
                                </div>
                                <div id="cont-size">
                                    <h4>{t('View the latest message updates about your patients.')}</h4>
                                </div>
                                <div id="cont-sized">
                                    <Link to={PREGNANCY_URL}>
                                        <button className="button-style default">{t('View')}</button>
                                    </Link>
                                </div>
                            </div>
                        </div>
                        <div className="spacer">
                            <img src={nbcPncAvatar} id="nbcPncAvatar" alt="NBC &#38; PNC module avatar" />

                            <div className="sub-container-message">
                                <div id="cont-size">
                                    <h1>{t('NBC & PNC')}</h1>
                                </div>
                                <div id="cont-size">
                                    <h4>{t('View the latest message updates about your patients.')}</h4>
                                </div>
                                <div id="cont-sized">
                                    <Link to={NBC_AND_PNC_URL}>
                                        <button className="button-style default">{t('View')}</button>
                                    </Link>
                                </div>
                            </div>
                        </div>
                        <div className="spacer">
                            <img src={nutritionAvatar} id="nutritionAvatar" alt="nutrition module avatar" />

                            <div className="sub-container-message">
                                <div id="cont-size">
                                    <h1>{t('Nutrition')}</h1>
                                </div>
                                <div id="cont-size">
                                    <h4>{t('View the latest message updates about your patients.')}</h4>
                                </div>
                                <div id="cont-sized">
                                    <Link to={NUTRITION_URL}>
                                        <button className="button-style default">{t('View')}</button>
                                    </Link>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

export default withTranslation()(Home);
