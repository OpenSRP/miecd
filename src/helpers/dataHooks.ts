import {
    OPENSRP_SECURITY_AUTHENTICATE,
    LOCATION_FILTER_PARAM,
    RISK_CATEGORY_FILTER_PARAM,
    SMS_TYPE_FILTER_PARAM,
    SEARCH_FILTER_PARAM,
    PREGNANCY_DETECTION,
    UPDATE_PREGNANCY_DETECTION,
    NUTRITION_REGISTRATION,
    UPDATE_NUTRITION_REGISTRATION,
    DEATH_REPORT,
    UPDATE_DEATH_REPORT,
} from '../constants';
import { getQueryParams } from 'containers/LogFace/utils';
import { TFunction } from 'i18next';
import { useQuery } from 'react-query';
import { useSelector } from 'react-redux';
import { RouteComponentProps } from 'react-router';
import { OpenSRPService } from 'services/opensrp';
import { getSmsDataByFilters } from 'store/ducks/sms_events';
import { TreeNode } from './locationHierarchy/types';
import { generateJurisdictionTree } from './locationHierarchy/utils';
import { toastToError, getRiskCatFilter } from './utils';
import { LogFaceModules } from 'configs/settings';

const selectSmsData = getSmsDataByFilters();

/** hook to get this users assignment details, hierarchy, team assigned and location the team is assigned
 * @param t - the translator function
 */
export const useUserAssignment = (t: TFunction) => {
    return useQuery(
        'userAssignment',
        () => {
            const opensrpService = new OpenSRPService(OPENSRP_SECURITY_AUTHENTICATE);
            return opensrpService.read('');
        },
        {
            select: (response) => {
                return {
                    userHierarchy: generateJurisdictionTree(response.locations),
                    user: response.user,
                    team: response.team.team,
                    location: response.team.team.location,
                };
            },
            onError: (err: Error) => toastToError(err?.message ?? t('An error occurred')),
        },
    );
};

/** redux hok to select filtered sms data based on provided filters
 * @param location - route location details, to get search params
 * @param module - the module
 * @param userLocationTree - for filtering based on the user's assigned level
 */
export const useGetLogFaceFilteredSms = (
    location: RouteComponentProps['location'],
    module: LogFaceModules,
    userHierarchy?: TreeNode,
) => {
    return useSelector((state) => {
        const userLocationIdFilter = getQueryParams(location)[LOCATION_FILTER_PARAM] as string;
        const riskCategoryFilter = getQueryParams(location)[RISK_CATEGORY_FILTER_PARAM] as string;
        const smsTypeFilter = getQueryParams(location)[SMS_TYPE_FILTER_PARAM] as string;
        const searchFilter = getQueryParams(location)[SEARCH_FILTER_PARAM] as string;
        const matchesId = (node: TreeNode) => node.model.id === userLocationIdFilter;
        const userLocationNode = userHierarchy?.first(matchesId);
        let smsFilter = smsTypeFilter ? [smsTypeFilter] : undefined;
        switch (smsTypeFilter) {
            case PREGNANCY_DETECTION:
                smsFilter = [PREGNANCY_DETECTION, UPDATE_PREGNANCY_DETECTION];
                break;
            case NUTRITION_REGISTRATION:
                smsFilter = [NUTRITION_REGISTRATION, UPDATE_NUTRITION_REGISTRATION];
                break;
            case DEATH_REPORT:
                smsFilter = [DEATH_REPORT, UPDATE_DEATH_REPORT];
                break;
            default:
                break;
        }
        return selectSmsData(state, {
            locationNode: userLocationNode,
            riskCategory: getRiskCatFilter(module, riskCategoryFilter),
            smsTypes: smsFilter,
            searchFilter,
            module,
        });
    });
};
