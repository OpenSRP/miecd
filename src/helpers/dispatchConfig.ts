import { ConfigState, setAllConfigs, getAllConfigs } from '@opensrp/pkg-config';
import { OPENSRP_REST_API_BASE_URL } from 'configs/settings';
import { BACKEND_ACTIVE } from '../configs/env';
import { BACKEND_LOGIN_URL, REACT_LOGIN_URL } from '../constants';
import i18n from '../mls';

export const APP_LOGIN_URL = BACKEND_ACTIVE ? BACKEND_LOGIN_URL : REACT_LOGIN_URL;

const defaultvalues = getAllConfigs();

const configObject: ConfigState = {
    ...defaultvalues,
    appLoginURL: APP_LOGIN_URL,
    opensrpBaseURL: OPENSRP_REST_API_BASE_URL,
    i18n,
};

setAllConfigs(configObject);
