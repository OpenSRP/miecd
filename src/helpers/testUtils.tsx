import { mount, MountRendererProps } from 'enzyme';
import React from 'react';
import { I18nextProvider } from 'react-i18next';
import i18n from '../mls';
import { act } from 'react-dom/test-utils';

export const mountWithTranslations: typeof mount = (component: React.ReactNode, options?: MountRendererProps) => {
    return mount(<I18nextProvider i18n={i18n}>{component}</I18nextProvider>, options);
};

/**
 * Utility to await promises to resolve in tests
 * @param ms time in ms
 */
export async function waitForPromises() {
    await act(async () => {
        await new Promise((res) => setImmediate(res));
    });
}
