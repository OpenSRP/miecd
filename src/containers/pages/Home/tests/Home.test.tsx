import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Router } from 'react-router';
import Home from '..';
import { mountWithTranslations } from '../../../../helpers/testUtils';

const history = createBrowserHistory();

describe('containers/pages/Home', () => {
    beforeEach(() => {
        jest.resetAllMocks();
    });

    // eslint-disable-next-line jest/expect-expect
    it('renders without crashing', () => {
        shallow(
            <Router history={history}>
                <Home />
            </Router>,
        );
    });

    it('renders Home correctly', () => {
        const wrapper = mountWithTranslations(
            <Router history={history}>
                <Home />
            </Router>,
        );
        expect(toJson(wrapper.find('.home-main'))).toMatchSnapshot();
        wrapper.unmount();
    });
});
