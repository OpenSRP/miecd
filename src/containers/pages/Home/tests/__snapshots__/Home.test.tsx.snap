// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/pages/Home renders Home correctly 1`] = `
<div
  className="home-main"
>
  <div
    className="welcome-text"
  >
    <h1>
      Welcome to the MIECD dashboard
    </h1>
  </div>
  <div
    className="center-vertically"
  >
    <div
      className="components-list"
    >
      <div
        className="spacer"
      >
        <img
          alt="pregnancy module avatar"
          id="pregnancyAvatar"
          src="pregnancy.jpeg"
        />
        <div
          className="sub-container-message"
        >
          <div
            id="cont-size"
          >
            <h1>
              Pregnancy
            </h1>
          </div>
          <div
            id="cont-size"
          >
            <h4>
              View the latest message updates about your patients.
            </h4>
          </div>
          <div
            id="cont-sized"
          >
            <Link
              to="/pregnancy"
            >
              <LinkAnchor
                href="/pregnancy"
                navigate={[Function]}
              >
                <a
                  href="/pregnancy"
                  onClick={[Function]}
                >
                  <button
                    className="button-style default"
                  >
                    View
                  </button>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </div>
      <div
        className="spacer"
      >
        <img
          alt="NBC & PNC module avatar"
          id="nbcPncAvatar"
          src="pnc_nbc.jpeg"
        />
        <div
          className="sub-container-message"
        >
          <div
            id="cont-size"
          >
            <h1>
              NBC & PNC
            </h1>
          </div>
          <div
            id="cont-size"
          >
            <h4>
              View the latest message updates about your patients.
            </h4>
          </div>
          <div
            id="cont-sized"
          >
            <Link
              to="/nbc_and_pnc"
            >
              <LinkAnchor
                href="/nbc_and_pnc"
                navigate={[Function]}
              >
                <a
                  href="/nbc_and_pnc"
                  onClick={[Function]}
                >
                  <button
                    className="button-style default"
                  >
                    View
                  </button>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </div>
      <div
        className="spacer"
      >
        <img
          alt="nutrition module avatar"
          id="nutritionAvatar"
          src="nutrition.jpeg"
        />
        <div
          className="sub-container-message"
        >
          <div
            id="cont-size"
          >
            <h1>
              Nutrition
            </h1>
          </div>
          <div
            id="cont-size"
          >
            <h4>
              View the latest message updates about your patients.
            </h4>
          </div>
          <div
            id="cont-sized"
          >
            <Link
              to="/nutrition"
            >
              <LinkAnchor
                href="/nutrition"
                navigate={[Function]}
              >
                <a
                  href="/nutrition"
                  onClick={[Function]}
                >
                  <button
                    className="button-style default"
                  >
                    View
                  </button>
                </a>
              </LinkAnchor>
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
